lisa/target.py

```diff
@@ -516,6 +516,7 @@ def _init_target(self, kind, name, workdir, device, host,
                 device = 'DEFAULT'
 
             conn_settings['device'] = device
+            conn_settings['adb_as_root'] = (username == 'root')
```

Comment:
Sadly this doesn't seem to cope with devices you can't root:

```
File "/data/work/lisa/lisa/target.py", line 266, in __init__
    wait_boot_timeout=wait_boot_timeout,
  File "/data/work/lisa/lisa/target.py", line 595, in _init_target
    target.connect(check_boot_completed=wait_boot, timeout=wait_boot_timeout)
  File "/data/work/lisa/external/devlib/devlib/target.py", line 1212, in connect
    super(AndroidTarget, self).connect(timeout=timeout, check_boot_completed=check_boot_completed)
  File "/data/work/lisa/external/devlib/devlib/target.py", line 263, in connect
    self._connections[tid] = self.get_connection(timeout=timeout)
  File "/data/work/lisa/external/devlib/devlib/target.py", line 283, in get_connection
    return self.conn_cls(timeout=timeout, **self.connection_settings)  # pylint: disable=not-callable
  File "/data/work/lisa/external/devlib/devlib/utils/android.py", line 272, in __init__
    self.adb_root(enable=True)
  File "/data/work/lisa/external/devlib/devlib/utils/android.py", line 334, in adb_root
    raise TargetStableError(output)
devlib.exception.TargetStableError: adbd cannot run as root in production builds
```
Dunno if we want to handle that properly or not. If the action to root depends on an extra conf key, maybe not.

Reply:
You will need to set a different username if your device cannot be rooted. LISA defaults to `Target.__init__(username="root")`. We already used to run `target.adb_root(enable=True)` unconditionally for android targets, so currently working setups will keep working (which was probably not the case with yours)